cli/cluster/config.go

```diff
@@ -0,0 +1,206 @@
+package cluster
+
+import (
+	"fmt"
+
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	fmtPathNotMap   = "path %q is not a map"
+	fmtPathNotExist = "path %q does not exist"
+)
+
+// Config is a container for deserialized configuration.
+type Config struct {
+	// paths is a container implementation for the deserialized configuration.
+	//
+	// At this moment it is fully-compatible with yaml.v2 Marshal()/Unmarshal()
+	// functions. So it could be marshaled and unmarshaled directly to YAML.
+	// We may change it in the future.
+	paths any
```

Comment:
Does unmarshaling to go struct type have any drawbacks? Probably `findInstance` might be much simpler if we are working with structs and not `map[any]any`.

Reply:
The `Сonfig` provides a generic interface independent of any structure. As example, we don't have a common structure for a cluster config and an instance config.

In the future, nothing prevents us from marshaling `Config` into a some structure via helper functions.

Reply:
1. But in fact we have... Global, group, replicaset and instance configurations have the same format with minor exceptions.
2. We need to change all updated paths. If I forget to change some path in my code, I will get an error in the middle of the work, not on YAML parsing at the beginning.

Reply:
You also need to take into account that not only YAML could be a source of configuration (at that moment it is environment variables too, but I think this list may expand in the future (as example, JSON from etcd or something else, there is no Tarantool 3.0 release yet)).

If we implement the config as a concrete type, then this will at least complicate extensibility: nothing prevents a user from adding additional sections to the configuration on different scopes and we will need to catch it somehow + merging becomes not such a trivial task.

I like the idea to keep the `Config` like a layer between configuration sources YAML, but it could be a good idea to add functions to transform the `Config` into some structs. In any case, this may be required further for validation: https://www.notion.so/tarantool/tt-3-0-fbd40e724dce4a04919d1a35c506a65b

```
Требования
- Производит валидацию конфига
```

So it will look like:

environment, YAML etc -> Config -> SomeBaseClusterStruct.
environment, YAML etc -> Config -> SomeBaseInstanceStruct.

Reply:
I'm not talking about full structuring of the config. I would like to see the global structure:
 config -> group -> replicaset -> instance
which will make the code of the `findInstance` and other config traverse code much easier. I'm not going to play with paths in forEach function. I find it much easier to make a simple loop and `get` call on the level I'm interested in.

Reply:
> I find it much easier to make a simple loop and `get` call on the level I'm interested in.

Do we really need in practice levels other than `cluster` and `instance`? For example, why do you need the `groups`, `replicasets` levels, what do you want to do with it?

If we do not need other levels, then from the user's point of view it is easier to get a list of instances from a cluster config and iterate over it:

```range
cconfig, _ := GetClusterConfig(path)
for _, instanceName := range GetClusterIntances(cconfig) {
    instances := GetInstanceConfig(cconfig, instanceName)
}
```

Reply:
> If we do not need other levels, then from the user's point of view it is easier to get a list of instances from a cluster config and iterate over it:

From user point of view - yes.
But we must maintain the code within. I think it will be much easier to maintain triple `for` loop instead of `forEach` and path manipulations. And if I ever want to write something new, it will be easier, because an editor will show me possible variations after typing '.'. Hope, I will never need to use `forEach` to iterate over a config elements :)

Reply:
Updated.